gha.dist.yml

```diff
@@ -103,7 +103,7 @@ jobs:
 
       - name: PHPUnit tests
         if: ${{ always() }}
-        run: moodle-plugin-ci phpunit
+        run: moodle-plugin-ci phpunit --coverage-text --fail-on-warning
```

Comment:
To enable here in the dist/reference file `--coverage-text` ... I'm not sure if it's a good idea. Right now we are using `phpdbg` to generate code coverage and it's crashing badly (GHA only, AFAIK), specially with PHP 8.0 and up (I had to disable it in a number of repos, see for example https://github.com/moodlehq/moodle-local_codechecker/commit/8f8ca5d4b4bb241f18d765d145a129e630d3cf63).

So, until we move to pcov (that seems to work way more reliably), I'd suggest not to "promote" it in the dist file.

About having the `--fail-on-warning` proposed there, np, IMO.

Reply:
yes, makes sense. I've been struggling to run coverage tests on php8 in gitlab as well :(